File: 1671/Solution.java
class Solution {
    public int minimumMountainRemovals(int[] nums) {
        int n = nums.length;
        int[] l = new int[n];
        int[] r = new int[n];
        for(int i = 0; i < n; i++){
            l[i] = 1;
            for(int j = 0; j < i; j++){
                if(nums[i] > nums[j]){
                    l[i] = Math.max(l[i], l[j] + 1);
                }
            }
        }
        for(int i = n-1; i > 0; i--){
            r[i] = 1;
            for(int j = n-1; j > i; j--){
                if(nums[i] > nums[j]){
                    r[i] = Math.max(r[i], r[j] + 1);
                }
            }
        }
        int max = 0;
        for (int i = 1; i < n-1; i++){
            if(l[i-1] == l[i]) continue;
            if(i == n -2 && r[i] == r[i+1]) continue;
            max = Math.max(max, l[i] + r[i]);
        }
        return n - max +1;
    }
}

File: 1671/Solution.rs
impl Solution {
    pub fn minimum_mountain_removals(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut l = vec![1; n];
        let mut r = vec![1; n];

        for i in 0..n {
            for j in 0..i {
                if nums[i] > nums[j] {
                    l[i] = l[i].max(l[j] + 1);
                }
            }
        }

        for i in (0..n).rev() {
            for j in i+1..n {
                if nums[i] > nums[j] {
                    r[i] = r[i].max(r[j] + 1);
                }
            }
        }

        let mut max_len = 0;

        for i in 1..n-1 {
            if l[i] > 1 && r[i] > 1 {
                max_len = max_len.max(l[i] + r[i])
            }
        }

        (n as i32) - max_len + 1
    }
}
